trl/trainer/utils.py

```diff
@@ -69,43 +69,64 @@ class DataCollatorForCompletionOnlyLM(DataCollatorForLanguageModeling):
     """
 
     def __init__(
-        self,
-        response_template: str,
-        *args,
-        mlm: bool = False,
-        ignore_index: int = -100,
-        **kwargs,
+        self, instruction_template, assistant_template, *args, mlm: bool = False, ignore_index: int = -100, **kwargs
     ):
         super().__init__(*args, mlm=mlm, **kwargs)
-        self.response_template = response_template
+        self.instruction_template = instruction_template
+        self.assistant_template = assistant_template
         self.ignore_index = ignore_index
 
     def torch_call(self, examples: List[Union[List[int], Any, Dict[str, Any]]]) -> Dict[str, Any]:
         batch = super().torch_call(examples)
 
-        # The prompt ends with the response key plus a newline.  We encode this and then try to find it in the
-        # sequence of tokens.  This should just be a single token.
-        response_token_ids = self.tokenizer.encode(self.response_template, add_special_tokens=False)
+        assistant_token_ids = self.tokenizer.encode(self.assistant_template, add_special_tokens=False)
 
         labels = batch["labels"].clone()
```

Comment:
To avoid needlessly increasing memory usage, all the operation on `labels` can be done directly on `batch["labels"]` so that cloning is not necessary.

Reply:
Good point !